site/content/entry/citra-progress-report-2018-q2.md

```diff
@@ -0,0 +1,305 @@
+useful for streamers who might want to display game information on their
+streams.
+
+## Movie - game input recording and playback ([#2882](https://github.com/citra-emu/citra/pull/2882), [#3922](https://github.com/citra-emu/citra/pull/3922)) by [danzel](https://github.com/danzel) and [zhaowenlan1779](https://github.com/zhaowenlan1779)
+
+This feature, not to be confused with video/audio recording, implements a way
+to record game inputs and plays them back. Every time the hardware reads an
+input device (buttons, touch screen, accelerometer, gyroscope, c-stick) we
+copy (when recording) or replace (when playing) the values. Once playback
+has finished, we release control of inputs so that the user can take control.
+
+{{< figure src="/images/entry/citra-progress-report-2018-q2/movie.png"
+    title="UI options to record" >}}
+
+For this, danzel heavily took inspiration from dolphin's system and zhaowenlan1779
+wrote the UI for this feature. This method results in `.ctm` (citra TAS movie)
+files which are playble in Citra. To play a game's `.ctm` file you would need
+to have the game and also set system time to the same starting value as in
+the `.ctm`. This feature makes debugging bugs in games much easier, as it
+doesn't require us to go through the whole process manually again.
+
+## [Add system time configuration](https://github.com/citra-emu/citra/pull/4043) by [B3N30](https://github.com/b3n30)
+
+Whenever a game requested time information, Citra always used your PC's system
+time. This method had a couple of issues. Consider if you were playing a game
+at 200% speed for 2 hours. As its running at twice the max speed, playing for
+2 hours in real-time should reflect 4 hours simulated time. But, due to Citra
+sending your PC's time to games, it wasn't happening.
+
+{{< figure src="/images/entry/citra-progress-report-2018-q2/system-clock.png"
+    title="System Time setting (edited to show both options)" >}}
+
+Also, players might want to set Citra's time to a specific value to ensure
+consistency with TAS or with RNG(random number generator) or for getting
+time specific game events. Earlier you had to modify your PC's system time
+to achieve this but thanks to B3N30, you can now just set it in Citra's
+configuration window.
+
+## [Joystick Hot-plugging](https://github.com/citra-emu/citra/pull/4141) by [B3N30](https://github.com/b3n30)
+
+For a long time, Citra didn't support controller hot plugging. This meant that
+if by any chance your controller disconnected Citra would fail to detect it
+after re-connection, unless you restarted Citra. This caused a lot of
+frustration among users who preferred playing with controllers.
+
+Hot plugging (also called hot swapping) is the ability to add and remove
+devices to a computer system while the computer is running and have the
+operating system automatically recognize the change.
+
+Thanks to the efforts of B3N30 and [jroweboy](https://github.com/jroweboy)-
```

Comment:
Watch the dash here - either double dash, or `Thanks to the efforts of B3N30 and jroweboy (who researched SDL controller configuration), Citra now doesn't [...]`